packages/eslint-plugin-shopify/lib/rules/jquery-dollar-sign-reference.js

```diff
@@ -0,0 +1,152 @@
+var JQUERY_IDENTIFIERS = ['$', 'jQuery'];
+
+var nonChainingMethods = {
+  context: true,
+  get: true,
+  hasClass: true,
+  height: true,
+  index: true,
+  innerHeight: true,
+  innerWidth: true,
+  is: true,
+  offset: true,
+  outerHeight: true,
+  outerWidth: true,
+  position: true,
+  promise: true,
+  scrollLeft: true,
+  scrollTop: true,
+  serialize: true,
+  serializeArray: true,
+  size: true,
+  text: true,
+  toArray: true,
+  triggerHandler: true,
+  val: true,
+  width: true,
+
+  html: function(node) {
+    return node.arguments.length === 0;
+  },
+  data: function(node) {
+    return node.arguments.length < 2 && node.arguments[0] && node.arguments[0].type !== 'ObjectExpression';
+  },
+  prop: function(node) {
+    return node.arguments.length < 2 && node.arguments[0] && node.arguments[0].type !== 'ObjectExpression';
+  },
+  attr: function(node) {
+    return node.arguments.length < 2 && node.arguments[0] && node.arguments[0].type !== 'ObjectExpression';
+  },
+};
+
+var JQUERY_IDENTIFIER_REGEX = /^\$./;
+
+module.exports = function(context) {
+  function getFinalReferenceName(node) {
+    switch (node.type) {
+    case 'Identifier': return node.name;
+    case 'Literal': return node.value;
+    case 'MemberExpression': return getFinalReferenceName(node.property);
+    default: return null;
+    }
+  }
+
+  function isjQueryReference(node) {
+    var referenceName = getFinalReferenceName(node);
+    return referenceName != null && JQUERY_IDENTIFIER_REGEX.test(referenceName);
```

Comment:
`test` will happily reject a `null` input, so this could be shorter.  No big deal either way, though.